frontend/src/metabase/query_builder/components/expressions/ExpressionWidget.tsx

```diff
@@ -63,7 +63,8 @@ export const ExpressionWidget = (props: ExpressionWidgetProps): JSX.Element => {
 
   const helpTextTargetRef = useRef(null);
 
-  const isValidName = withName ? !!name : true;
+  const isOnlySpaces = (str: string) => /^\s*$/.test(str);
+  const isValidName = withName ? !!name && !isOnlySpaces(name) : true;
```

Comment:
1. This can be done without regexp and without the function
2. The `!!name` part is no longer necessary

```suggestion
  const isValidName = withName ? name.trim().length > 0 : true;
```